packages/components/src/TooltipTrigger/TooltipTrigger.component.js

```diff
@@ -24,124 +35,168 @@ const props = {
-			this.overlay.handleDelayedShow();
+		if (dimensions.left - tooltipWidth / 2 < 0) {
+			if (['top', 'bottom'].includes(tooltipPlacement)) {
+				placement = `${placement}-right`;
+			} else if (tooltipPlacement === 'left') {
+				placement = placement.replace('left', 'right');
+			}
+		} else if (dimensions.right + tooltipWidth / 2 > window.innerWidth) {
+			if (['top', 'bottom'].includes(tooltipPlacement)) {
+				placement = `${placement}-left`;
+			} else if (tooltipPlacement === 'right') {
+				placement = placement.replace('right', 'left');
+			}
 		}
-	};
+		return placement;
+	}
 
-	render() {
-		const child = React.Children.only(this.props.children);
+	getTooltipPosition() {
+		const { tooltipPlacement = 'right', tooltipWidth = DEFAULT_OFFSET_X } = this.props;
 
-		if (!this.state.hovered) {
-			return cloneElement(child, {
-				onMouseOver: this.onMouseOver,
-				onFocus: this.onFocus,
-			});
+		if (!this.el) {
+			return {
+				tooltipPlacement,
+			};
 		}
 
-		const tooltip = (
-			<Tooltip className={getTooltipClass()} id={this.state.id}>
-				{this.props.label}
-			</Tooltip>
-		);
-		// TODO jmfrancois : render the Tooltip in a provider so use context for that.
-		// fix with disabling element https://github.com/react-bootstrap/react-bootstrap/pull/3251
-		// we add onClick={() => this.overlay.handleDelayedHide() to hide the overlay
-		// it stays when the element inside is disabled
+		const dimensions = this.el.getBoundingClientRect();
+
+		const placement = this.getAdjustedTooltipPlacement(tooltipPlacement, dimensions);
+
+		return {
+			placement,
+			style: {
+				top: (() => {
+					if (placement.startsWith('bottom')) {
+						return dimensions.bottom;
+					}
+					if (['left', 'right'].includes(placement)) {
+						return (dimensions.top + dimensions.bottom) / 2;
+					}
+					return undefined;
+				})(),
+				left: (() => {
+					if (placement === 'top-right' || placement === 'bottom-right') {
+						return (dimensions.left + dimensions.right) / 2;
+					}
+					if (placement === 'top-left' || placement === 'bottom-left') {
+						return (dimensions.left + dimensions.right) / 2 - tooltipWidth;
+					}
+					if (placement.includes('left')) {
+						return `calc(${Math.trunc(dimensions.left)}px - ${Math.trunc(tooltipWidth)}px)`;
+					}
+					if (placement.includes('right')) {
+						return Math.trunc(dimensions.right);
+					}
+					if (['top', 'bottom'].includes(placement)) {
+						return (dimensions.left + dimensions.right) / 2;
+					}
+					return undefined;
+				})(),
+				bottom: (() => {
+					if (placement === 'top' || placement === 'top-right' || placement === 'top-left') {
+						return `calc(100vh - ${Math.trunc(dimensions.top)}px)`;
+					}
+					if (placement === 'right-bottom' || placement === 'left-bottom') {
+						return `calc(100vh - ${Math.trunc((dimensions.top + dimensions.bottom) / 2)}px)`;
+					}
+					return undefined;
+				})(),
+			},
+		};
+	}
+
+	showTooltip() {
```

Comment:
You could create a custom hook for the show / hide behavior.
And only allow an inital value, and a callback for each.
It will help to read and maintain the code to separate concern.

Reply:
Could you help to understand that please? 🙏